### ISSUE TITLE
Flawless PopOS Installation

### ISSUE THREAD

**Issue by mssalvatore** [guardicore/monkey#1380]
# Spike

## Objective
Test the installation of Infection Monkey on PopOS

## Scope
The below tests are not intended to be exhaustive. Rather, they are smoke tests to verify the basic functionality of Infection Monkey on PopOS.

- [ ] Infection monkey installs without errors or warnings.
- [ ] Monkey Island starts without errors or warning.
- [ ] No additional system changes (dependencies, settings, etc) are required to successfully install and run Monkey Island
- [ ] Monkey agent can be started locally
- [ ] Monkey agent successfully propagates to a victim machine
- [ ] Monkey Island receives telemetry and other expected data from monkey agents

## Output

The output of this spike is a set of GitHub issues (bugs). When these bugs are resolved, we should be able to claim to have a flawless installation experience on PopOS.

**Comment by TheCoreMan** [comment]
I'm gonna comment all issues I find

The installation instructions are out of date, refer `deb` and not `appimage` https://www.guardicore.com/infectionmonkey/docs/setup/debian/

**Comment by TheCoreMan** [comment]
chmod +x and running initial output:

```sh
❯ chmod +x Infection_Monkey.AppImage
❯ ./Infection_Monkey.AppImage
2021-08-13 01:32:41,431 - selector_events.py:58 -   __init__() - DEBUG - Using selector: GeventSelector
2021-08-13 01:32:41,665 - report_generation_synchronisation.py:12 -   <module>() - DEBUG - Initializing report generation locks.
2021-08-13 01:32:41,856 - environment_singleton.py:44 - initialize_from_file() - INFO - Monkey's env is: StandardEnvironment
2021-08-13 01:32:41,857 - mongo_setup.py:36 - _create_db_dir() - INFO - Database content directory: /home/shay/.monkey_island/db.
2021-08-13 01:32:41,857 - mongo_db_process.py:23 -      start() - INFO - Starting MongoDB process.
2021-08-13 01:32:41,857 - mongo_db_process.py:24 -      start() - DEBUG - MongoDB will be launched with command: /tmp/.mount_InfectjV4xvP/usr/src/monkey_island/bin/mongodb/bin/mongod --dbpath /home/shay/.monkey_island/db.
2021-08-13 01:32:41,857 - mongo_db_process.py:25 -      start() - INFO - MongoDB log will be available at /home/shay/.monkey_island/mongodb.log.
2021-08-13 01:32:41,864 - mongo_db_process.py:32 -      start() - INFO - MongoDB launched successfully!
2021-08-13 01:32:42,395 - mongo_setup.py:54 - _wait_for_mongo_db_server() - INFO - Waiting for MongoDB server on mongodb://localhost:27017/monkeyisland
2021-08-13 01:32:43,419 - mongo_setup.py:74 - _assert_mongo_db_version() - INFO - Mongo DB version OK. Got ('4', '2', '3')
2021-08-13 01:32:43,439 - connectionpool.py:230 -  _new_conn() - DEBUG - Starting new HTTP connection (1): 169.254.169.254:80
2021-08-13 01:32:45,444 - aws_instance.py:44 -   __init__() - DEBUG - Failed init of AwsInstance while getting metadata: HTTPConnectionPool(host='169.254.169.254', port=80): Max retries exceeded with url: /latest/meta-data/instance-id (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7ff089226350>, 'Connection to 169.254.169.254 timed out. (connect timeout=2)'))
2021-08-13 01:32:45,451 - connectionpool.py:230 -  _new_conn() - DEBUG - Starting new HTTP connection (1): 169.254.169.254:80
2021-08-13 01:32:46,520 - aws_instance.py:54 -   __init__() - DEBUG - Failed init of AwsInstance while getting dynamic instance data: HTTPConnectionPool(host='169.254.169.254', port=80): Max retries exceeded with url: /latest/dynamic/instance-identity/document (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff089226f50>: Failed to establish a new connection: [Errno 113] No route to host'))
2021-08-13 01:32:46,527 - auth.py:22 -   init_jwt() - DEBUG - Initialized JWT with secret key that started with c66e
2021-08-13 01:32:46,577 - database_initializer.py:13 - init_collections() - INFO - Setting up the Monkey Island, this might take a while...
2021-08-13 01:32:46,734 - server_setup.py:121 - _start_island_server() - INFO - Using certificate path: /tmp/.mount_InfectjV4xvP/usr/src/monkey_island/cc/server.crt, and key path: /tmp/.mount_InfectjV4xvP/usr/src/monkey_island/cc/server.key.
2021-08-13 01:32:46,734 - server_setup.py:155 - _log_init_info() - INFO - Monkey Island Server is running!
2021-08-13 01:32:46,734 - server_setup.py:156 - _log_init_info() - INFO - version: 1.10.0+dev
2021-08-13 01:32:46,741 - server_setup.py:162 - _log_init_info() - INFO - Listening on the following URLs: https://192.168.253.16:5000, https://172.17.0.1:5000, https://172.18.0.1:5000, https://192.168.1.230:5000
2021-08-13 01:32:46,873 - monkey_download.py:114 - log_executable_hashes() - DEBUG - monkey-windows-32.exe hashes:
SHA-256 170b884d7c92f279fea1c8c558022815d5dc9db6a8a4dbf234799c39f5989b32
2021-08-13 01:32:46,972 - monkey_download.py:114 - log_executable_hashes() - DEBUG - monkey-linux-32 hashes:
SHA-256 f45a696d23b8d31e89d44baede57c177250db005b74ae8d0d82f79413ae125fe
2021-08-13 01:32:47,072 - monkey_download.py:114 - log_executable_hashes() - DEBUG - monkey-linux-64 hashes:
SHA-256 cc1dbf9a579e192c31990362662e6f16bb278c1090db77e3dc61d2d7aa45c3d7
2021-08-13 01:32:47,187 - monkey_download.py:114 - log_executable_hashes() - DEBUG - monkey-windows-64.exe hashes:
SHA-256 ac00c5b3458277df9549e925a2546ae860123a9a743174fe107c8b2629220805
```

**Comment by TheCoreMan** [comment]
OS info:

```
❯ uname -a
Linux pain-os 5.11.0-7620-generic #21~1626191760~21.04~55de9c3-Ubuntu SMP Tue Jul 20 22:18:55 UTC  x86_64 x86_64 x86_64 GNU/Linux
❯ lsb_release --all
No LSB modules are available.
Distributor ID: Pop
Description:    Pop!_OS 21.04
Release:        21.04
Codename:       hirsute
```

**Comment by TheCoreMan** [comment]
Error when clicking around in configuration. Went to the Ransomware scenario, then to configure monkey. Set some settings, went to "Randomware", and got this:

![image](https://user-images.githubusercontent.com/6576891/129278633-3f3155ab-54c2-4f04-9cd3-b95935d2b59e.png)

Let me know when you're ready for another test @mssalvatore @ofriz

**Comment by TheCoreMan** [comment]
api/island-mode:1 Failed to load resource: the server responded with a status of 401 (UNAUTHORIZED)
AuthService.js:87 Got 401 from server while trying to authFetch: {"error":"Authentication error: Missing Authorization Header"}
react-dom.production.min.js:209 TypeError: Cannot read property 'encryption' of undefined
    at Array.r (UISchemaManipulators.tsx:17)
    at t.default (UISchemaManipulators.tsx:8)
    at f.renderConfigContent (ConfigurePage.js:449)
    at f.value (ConfigurePage.js:509)
    at La (react-dom.production.min.js:182)
    at Na (react-dom.production.min.js:181)
    at vs (react-dom.production.min.js:263)
    at lc (react-dom.production.min.js:246)
    at sc (react-dom.production.min.js:246)
    at $s (react-dom.production.min.js:239)
es @ react-dom.production.min.js:209
react-dom.production.min.js:123 Uncaught TypeError: Cannot read property 'encryption' of undefined
    at Array.r (UISchemaManipulators.tsx:17)
    at t.default (UISchemaManipulators.tsx:8)
    at f.renderConfigContent (ConfigurePage.js:449)
    at f.value (ConfigurePage.js:509)
    at La (react-dom.production.min.js:182)
    at Na (react-dom.production.min.js:181)
    at vs (react-dom.production.min.js:263)
    at lc (react-dom.production.min.js:246)
    at sc (react-dom.production.min.js:246)
    at $s (react-dom.production.min.js:239)

**Comment by TheCoreMan** [comment]
Server logs
```
2021-08-13 01:34:55,333 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:55,336 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:58,441 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:58,443 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:58,456 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:58,463 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:58] "GET / HTTP/1.1" 200 840 0.003225
2021-08-13 01:34:58,567 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:58] "GET /main.js HTTP/1.1" 200 3820504 0.015747
2021-08-13 01:34:58,845 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:58] "GET /api/island-mode HTTP/1.1" 401 181 0.000642
2021-08-13 01:34:59,098 - auth.py:72 - _create_access_token() - DEBUG - Created access token for user 1234567890!@#$%^&*()_nothing_up_my_sleeve_1234567890!@#$%^&*() that begins with eyJ02021-08-13 01:34:59,098 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:59] "POST /api/auth HTTP/1.1" 200 408 0.2416362021-08-13 01:34:59,103 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:59] "GET /fe28785e4b24a4fcde829dc6942a9c27.gif HTTP/1.1" 200 169928 0.002083
2021-08-13 01:34:59,103 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']2021-08-13 01:34:59,137 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']2021-08-13 01:34:59,137 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:59,138 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:59,139 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:59,153 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:59] "GET /815d861e76071f51f82bcc2c77f300be.svg HTTP/1.1" 200 6080 0.000955
2021-08-13 01:34:59,154 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:59] "GET /b30dd0811bb67a69d1549c86bcc2fd2b.png HTTP/1.1" 200 35515 0.001052
2021-08-13 01:34:59,156 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:59] "GET /ef442c18f2fa5b0bffbd88501e6f2624.svg HTTP/1.1" 200 7135 0.001632
2021-08-13 01:34:59,157 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:34:59,158 - connectionpool.py:943 -  _new_conn() - DEBUG - Starting new HTTPS connection (1): updates.infectionmonkey.com:443
2021-08-13 01:34:59,160 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:59] "GET /api/island-mode HTTP/1.1" 200 122 0.011597
2021-08-13 01:34:59,398 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:34:59] "GET /favicon.ico HTTP/1.1" 200 110810 0.001884
2021-08-13 01:35:01,672 - connectionpool.py:442 - _make_request() - DEBUG - https://updates.infectionmonkey.com:443 "GET /?deployment=standard&monkey_version=1.10.0+dev HTTP/1.1" 200 61
2021-08-13 01:35:01,674 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:01] "GET /api/version-update HTTP/1.1" 200 285 2.524648
2021-08-13 01:35:01,675 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:01] "GET /api/version-update HTTP/1.1" 200 285 0.000438
2021-08-13 01:35:09,565 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:09] "GET /api/island-mode HTTP/1.1" 200 122 0.001799
2021-08-13 01:35:15,780 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:15,805 - config.py:207 - update_config() - INFO - monkey config was updated
2021-08-13 01:35:15,805 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:15] "POST /api/island-mode HTTP/1.1" 200 110 0.022621
2021-08-13 01:35:15,979 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:15] "GET /api/version-update HTTP/1.1" 200 285 0.000560
2021-08-13 01:35:15,980 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:15,983 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:15] "GET /api/island-mode HTTP/1.1" 200 130 0.002420
2021-08-13 01:35:16,081 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:16] "GET /api/version-update HTTP/1.1" 200 285 0.000552
2021-08-13 01:35:16,087 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:16,090 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:16,106 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:16] "GET /815d861e76071f51f82bcc2c77f300be.svg HTTP/1.1" 200 6080 0.000776
2021-08-13 01:35:16,107 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:16] "GET /ef442c18f2fa5b0bffbd88501e6f2624.svg HTTP/1.1" 200 7135 0.000694
2021-08-13 01:35:16,108 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:16,110 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:16] "GET /api HTTP/1.1" 200 413 0.017171
2021-08-13 01:35:18,872 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:18,891 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:18] "GET /api/island-mode HTTP/1.1" 200 130 0.009049
2021-08-13 01:35:18,911 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:18] "GET /api HTTP/1.1" 200 413 0.005552
2021-08-13 01:35:25,294 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:25,300 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:25,303 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:25] "GET /api/configuration/island HTTP/1.1" 200 32169 0.004398
2021-08-13 01:35:25,308 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:25,310 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:25] "GET /api/configuration/island HTTP/1.1" 200 32169 0.004027
2021-08-13 01:35:25,313 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:25,317 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:25] "GET /api/configuration/island HTTP/1.1" 200 32169 0.002618
2021-08-13 01:35:25,319 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:25] "GET /api/local-monkey HTTP/1.1" 200 129 0.017915
2021-08-13 01:35:25,322 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:25,326 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:25] "GET /api/local-monkey HTTP/1.1" 200 129 0.004459
2021-08-13 01:35:28,872 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:28,891 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:28] "GET /api/island-mode HTTP/1.1" 200 130 0.008617
2021-08-13 01:35:28,905 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:28] "GET /api HTTP/1.1" 200 413 0.003404
2021-08-13 01:35:29,173 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:29] "GET /api/configuration/island HTTP/1.1" 200 32169 0.004131
2021-08-13 01:35:29,174 - gevent_hub_error_handler.py:25 -   __call__() - WARNING - gevent caught an exception: ['ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:1091)\n']
2021-08-13 01:35:29,190 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:29] "GET /api/attack HTTP/1.1" 200 12217 0.013704
2021-08-13 01:35:29,344 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:29] "GET /412a43d6840addd683665ec12c30f810.woff2 HTTP/1.1" 200 80582 0.001678
2021-08-13 01:35:38,874 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:38] "GET /api/island-mode HTTP/1.1" 200 130 0.009465
2021-08-13 01:35:38,903 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:38] "GET /api HTTP/1.1" 200 413 0.002434
2021-08-13 01:35:47,988 - pywsgi.py:1226 -      write() - INFO - 192.168.253.16 - - [2021-08-13 01:35:47] "GET /main.js.map HTTP/1.1" 200 11160739 0.044175
```

**Comment by mssalvatore** [comment]
This appears to be resolved in Infection Monkey v2.2.0 on PopOS 22.04. @TheCoreMan We can reopen this if you have a different experience.
